go-controller/pkg/ovn/secondary_layer2_network_controller_test.go

```diff
@@ -380,12 +380,17 @@ var _ = Describe("OVN Multi-Homed pod operations for layer2 network", func() {
 				By("asserting the OVN entities provisioned in the NBDB are the expected ones after migration")
 				expectedPodLspEnabled := map[string]*bool{}
 				expectedPodLspEnabled[sourcePodInfo.podName] = migrationInfo.sourcePodInfo.expectedLspEnabled
-				expectedPodLspEnabled[targetPodInfo.podName] = migrationInfo.targetPodInfo.expectedLspEnabled
+
+				testPods := []testPod{sourcePodInfo}
+				if !util.PodCompleted(targetKvPod) {
+					testPods = append(testPods, targetPodInfo)
+					expectedPodLspEnabled[targetPodInfo.podName] = migrationInfo.targetPodInfo.expectedLspEnabled
+				}
```

Comment:
this makes me believe we should move all the `expectedPodLspEnabled` and `testPods` logic into an helper.

Maybe a struct that has the 2 pods as parameters and spits out the expectedEnabledLSPs / relevant pods for the test ?

Reply:
I don't follow.. Can you elaborate?

Reply:
Add something like the following:
```go
type migrationEntities struct {
    srcPod *v1.Pod
    dstPod *v1.Pod
}

func (me *migrationEntities) expectedLSPs() []LSP {
    return lsps
}

func (me *migrationEntities) pods () []*v1.Pod {
    return []*v1.Pod{me.srcPod, me.dstPod}
}
```

Something like that. 

Maybe it's overkill, but it can help to move away this distracting logic.

Reply:
well there is a struct I made for the live-migration pods `liveMigrationPodInfo` that holds the `expectedLspEnabled` parameter, but it lives only for the sake of this `DescribeTest`..
Do I think it makes sense to add a method for it? I honestly don't think so..

Reply:
Let's forget about it then.